File: explicit-python/explicit/__test__/ExplicitEngineTest.py
import unittest

from explicit.ExplicitEngine import ExplicitEngine
from explicit.parser.xml.XmlExplicitRulesParser import XmlExplicitRulesParser


class ExplicitEngineTest(unittest.TestCase):

    def test_simple_string_extractor(self):
        rulez = XmlExplicitRulesParser().parse("./extractors.xml")
        engine = ExplicitEngine(rulez)
        result = engine.execute("simple string extractor")

        self.assertEqual([{'one': 'A', 'two': 'B'}], result)

    def test_simple_number_extractor(self):
        rulez = XmlExplicitRulesParser().parse("./extractors.xml")
        engine = ExplicitEngine(rulez)
        result = engine.execute("simple number extractor")

        self.assertEqual([{'one': 1, 'two': 2}], result)

    def test_simple_boolean_extractor(self):
        rulez = XmlExplicitRulesParser().parse("./extractors.xml")
        engine = ExplicitEngine(rulez)
        result = engine.execute("simple boolean extractor")

        self.assertEqual([{'one': True, 'two': False}], result)

    def test_simple_variable_extractor(self):
        rulez = XmlExplicitRulesParser().parse("./extractors.xml")
        engine = ExplicitEngine(rulez)
        result = engine.execute("simple variable extractor")

        self.assertEqual([{'one': "variable", 'two': "extractor"}], result)

    def test_simple_function_extractor(self):
        rulez = XmlExplicitRulesParser().parse("./extractors.xml")
        engine = ExplicitEngine(rulez)
        result = engine.execute("simple function extractor")

        self.assertEqual([{'one': "FUNCTION", 'two': "EXTRACTOR"}], result)

    if __name__ == '__main__':
        unittest.main()

File: explicit-java/src/test/kotlin/explicit/ExplicitEngineTest.kt
package explicit

import explicit.parser.xml.ExplicitXmlParser
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class ExplicitEngineTest {

    @Test
    fun `simple string extractor`() {
        val stream = this::class.java.getResourceAsStream("/extractors.xml")
        val rulez = ExplicitXmlParser().parse(stream)

        val engine = ExplicitEngine(rulez)
        val result = engine.execute("simple string extractor")
        Assertions.assertEquals("[{one=A, two=B}]", result.toString())
    }

    @Test
    fun `simple number extractor`() {
        val stream = ExplicitXmlParser::class.java.getResourceAsStream("/extractors.xml")
        val rulez = ExplicitXmlParser().parse(stream)

        val engine = ExplicitEngine(rulez)
        val result = engine.execute("simple number extractor")
        Assertions.assertEquals("[{one=1, two=2}]", result.toString())
    }

    @Test
    fun `simple boolean extractor`() {
        val stream = ExplicitXmlParser::class.java.getResourceAsStream("/extractors.xml")
        val rulez = ExplicitXmlParser().parse(stream)

        val engine = ExplicitEngine(rulez)
        val result = engine.execute("simple boolean extractor")
        Assertions.assertEquals("[{one=true, two=false}]", result.toString())
    }

    @Test
    fun `simple variable extractor`() {
        val stream = ExplicitXmlParser::class.java.getResourceAsStream("/extractors.xml")
        val rulez = ExplicitXmlParser().parse(stream)

        val engine = ExplicitEngine(rulez)
        val result = engine.execute("simple variable extractor")
        Assertions.assertEquals("[{one=variable, two=extractor}]", result.toString())
    }

    @Test
    fun `simple function extractor`() {
        val stream = ExplicitXmlParser::class.java.getResourceAsStream("/extractors.xml")
        val rulez = ExplicitXmlParser().parse(stream)

        val engine = ExplicitEngine(rulez)
        val result = engine.execute("simple function extractor")
        Assertions.assertEquals("[{one=FUNCTION, two=EXTRACTOR}]", result.toString())
    }

}
